core/src/main/java/software/amazon/awssdk/auth/ContainerCredentialsProvider.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.auth;
+
+import static java.util.stream.Collectors.joining;
+import static software.amazon.awssdk.utils.FunctionalUtils.invokeSafely;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.AwsSystemSetting;
+import software.amazon.awssdk.SdkClientException;
+import software.amazon.awssdk.internal.CredentialsEndpointProvider;
+import software.amazon.awssdk.retry.internal.CredentialsEndpointRetryPolicy;
+import software.amazon.awssdk.utils.SdkAutoCloseable;
+import software.amazon.awssdk.utils.StringUtils;
+import software.amazon.awssdk.utils.Validate;
+
+/**
+ * {@link AwsCredentialsProvider} implementation that loads credentials from the Amazon Elastic Container Service.
+ *
+ * <p>The URI path is retrieved from the environment variable "AWS_CONTAINER_CREDENTIALS_RELATIVE_URI" in the container's
+ * environment. If the environment variable is not set, this credentials provider will always return null.</p>
```

Comment:
Shouldn't individual providers throw an exception?

Reply:
Good catch, that is what the logic does it just wasn't documented correctly.